### ISSUE TITLE
Alternate Design Documentation for Projectile

### ISSUE THREAD

**Issue by smiths** [JacquesCarette/Drasil#1681]
The documentation in #1648 shows a Parnas inspired MG/MIS.  There are some things about this approach that are in its favour, but it might be closer to the code than we want.  We might even want to generate the MIS (or something similar) from higher level design decisions.

Are any design decisions actually made in Projectile?  There is `Bundled` versus `Unbundled`.  Are there any other decisions?  Is there any new information?

My thinking was to write a design document that uses the infrastructure for the SRS.  We just shift the how and the what.  What was previously left abstract is filled in.  The idea is to continue the refinements that we had for the SRS.  The idea is to consider what comes after the instance model.  What is the next refinement?

@samm82, can you think about what such a document would look like?  I'm worried that there actually isn't any refinement.  All of the information that we need may already be in the SRS for Projectile.  In GlassBR we add the notion of an interpolation algorithm.  In SWHS we add the notion of an ODE solver, but Projectile is a set of explicit equations that are solved by substitution.

In addition to looking for additional algorithm information, I would also like you to brainstorm on design decision variabilities.  You will find them in the input and output format and in the organization of the code.

We can discuss this further during our meeting on Thursday.

**Comment by samm82** [comment]
I think the only other design decisions in Projectile pertain to constraints. There is an option to make violated constraints throw warnings or exceptions. Right now, they are set to warnings in the generated code, and exceptions in the MIS. I'm not sure which direction you wanted to take with this. I also assumed that we'd generate different design documentation for the `Bundled` or `Unbundled` code (and the constraints setting, for that matter), but let me know if I understand that wrong.

I'm not exactly sure what you have in mind for this new design document, but I'm pretty sure it would be generated straight from the knowledge captured for the SRS. I don't see any place where refinement would be necessary/useful/ The only real refinement that I can see is how modules are grouped, and that is already covered in the MIS.

The current variabilities I've come up with can be found [in the Projectile project](https://github.com/JacquesCarette/Drasil/projects/5#card-21914845). Would the organization of the code variabilities be along the lines of `Bundled` vs. `Unbundled`? I'm not sure what variabilities would be desired in that category.

**Comment by bmaclach** [comment]
There's also the logging option, which is another design variability that we've already implemented.

I too was under the impression that we'd just generate different design documentation corresponding to the design decisions made, so the MIS being specific to the `Bundled` design decision made sense to me.

**Comment by smiths** [comment]
Yes, we would generate the design documentation based on the design decisions.  I did not phrase that well when I mentioned it above.  What I'm looking for is how we record our design decisions.  I guess bigger than that, I'm looking for what design decisions we are making between the SRS and the code.  

We should think of the MIS as a generated document.  (I slipped a bit on this in my previous thinking.)  The question then is, does it serve a useful purpose, especially for an example like Projectile.  I was looking at the design stage to make decisions that are necessary for GOOL.  If the Projectile MIS is generated from the SRS, does the abstraction it represents provide anything useful that is no in the code?  It would be a language agnostic version of the code.  Maybe that is useful?

We need to determine what design decisions we are making, where we make them in the process, and where we document them.

We can discuss further during our meeting on Thursday.
